semantics/cpp14/language/translation/operators.k

```diff
@@ -26,6 +26,9 @@ module CPP-OPERATORS
           => builtinOperator(O, E1)
           requires notBool isCPPClassType(T1) andBool notBool isCPPEnumType(T1)
 
+     rule #UnaryOperator(operator&, E1::Expr, T1::CPPEnumType)
```

Comment:
This rule is not correct. In the case where T1 is an enumeration type, `#UnaryOperator` needs to perform overload resolution to determine whether an overloaded operator exists. This happens by computing the candidate sets according to paragraph 13.3.1.2:3. You don't have to implement the full generality of operator overloading, however, you need to do at least the following:

1. Compute the set of member candidates to be empty because T1 is not a class type.
2. Perform unqualified name lookup on the operator in order to compute the candidate set of non-member candidates. This needs to include arg dependent name lookup as well.
3. Compute the set of built-in candidates for the unary operator & as the empty set.
4. If all three candidate sets are empty, **then** you can select builtinOperator(operator&, E1).